source/client/process_impl.cc

```diff
@@ -338,6 +338,71 @@ getUserDefinedFactoryConfigPairs(const Options& options) {
   return factory_config_pairs;
 }
 
+/**
+ * Takes in a single worker's user defined outputs and collects them into the
+ * user_defined_outputs_by_plugin map. When called for each worker, transforms the outputs from
+ * being mapped by Worker to being mapped by Plugin.
+ *
+ * @param user_defined_outputs_by_plugin The map that this function will collect the worker data
+ * into, maps plugin name to the set of results for that plugin.
+ * @param worker_user_defined_outputs The per worker results to collect and organize.
+ */
+void collectUserDefinedOutputs(
+    absl::flat_hash_map<std::string, std::vector<nighthawk::client::UserDefinedOutput>>*
+        user_defined_outputs_by_plugin,
+    const std::vector<nighthawk::client::UserDefinedOutput>& worker_user_defined_outputs) {
+  for (const nighthawk::client::UserDefinedOutput& user_defined_output :
+       worker_user_defined_outputs) {
+    std::vector<nighthawk::client::UserDefinedOutput> cross_worker_results_for_plugin =
+        (user_defined_outputs_by_plugin->contains(user_defined_output.plugin_name())
+             ? user_defined_outputs_by_plugin->find(user_defined_output.plugin_name())->second
+             : std::vector<nighthawk::client::UserDefinedOutput>{});
+    cross_worker_results_for_plugin.emplace_back(user_defined_output);
+    user_defined_outputs_by_plugin->insert_or_assign(user_defined_output.plugin_name(),
```

Comment:
If I understand the code correctly, we make a local copy of the vector from the user_defined_outputs_by_plugin, then push a new entry at its back. Then we assign it back to the map, throwing away the original vector it used to hold. That is going to be suboptimal.

The bigger problem is that it feels unintuitive, it took me a considerable amount of time to understand this code (assuming I got what it does).

Can we try to simplify this? If I understand what the code wants to do, something like this (pseudocode) would be more idiomatic:

```cc
if (!user_defined_outputs_by_plugin.contains(plugin_name)) {
  user_defined_outputs_by_plugin[plugin_name] = {user_defined_output}
} else {
  user_defined_outputs_by_plugin[plugin_name].push_back(user_defined_output)
}
```